render/go/renderer_test.go

```diff
@@ -0,0 +1,181 @@
+package golang_test
+
+import (
+	"github.com/sasswart/gin-in-a-can/openapi"
+	"github.com/sasswart/gin-in-a-can/openapi/media"
+	"github.com/sasswart/gin-in-a-can/openapi/request"
+	"github.com/sasswart/gin-in-a-can/openapi/schema"
+	"github.com/sasswart/gin-in-a-can/render"
+	golang "github.com/sasswart/gin-in-a-can/render/go"
+	"github.com/sasswart/gin-in-a-can/test"
+	"github.com/sasswart/gin-in-a-can/tree"
+	"net/http"
+	"testing"
+)
+
+func TestGolang_SetTemplateFuncMap(t *testing.T) {
+	g := &golang.Renderer{Base: &render.Base{}}
+	g.SetTemplateFuncMap(nil)
+	if g.Base.TemplateFuncMapping == nil {
+		t.Errorf("TemplateFuncMapping error")
+	}
+	if g.GetTemplateFuncMap() == nil {
+		t.Errorf("GetTemplateFuncMap() error")
+	}
+}
+
+func TestGolang_SanitiseName(t *testing.T) {
+
+	specPath := "../../" + test.OpenapiFile
+	apiSpec, _ := openapi.LoadAPISpec(specPath)
+
+	tests := []struct {
+		name     string
+		expected string
+		node     tree.NodeTraverser
+	}{
+		{
+			name:     "testopenapi root",
+			node:     apiSpec,
+			expected: "ValidationFixture",
+		},
+		{
+			name:     "testopenapi path",
+			node:     test.Dig(apiSpec, test.Endpoint),
+			expected: "ValidationFixtureEndpoint",
+		},
+		{
+			name:     "testopenapi pathitem",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost),
+			expected: "ValidationFixtureEndpointPost",
+		},
+		{
+			name:     "testopenapi parameter",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost, "0"),
+			expected: "Parameter",
+		},
+		{
+			name:     "testopenapi requestbody",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost, request.BodyKey),
+			expected: "ValidationFixtureEndpointPostRequestBody",
+		},
+		{
+			name:     "testopenapi json mediaitem",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost, request.BodyKey, media.JSONKey),
+			expected: "ValidationFixtureEndpointPostRequestBody",
+		},
+		{
+			name:     "testopenapi schema",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost, request.BodyKey, media.JSONKey, schema.Key),
+			expected: "ValidationFixtureEndpointPostRequestBodyModel",
+		},
+		{
+			name:     "testopenapi property",
+			node:     test.Dig(apiSpec, test.Endpoint, http.MethodPost, request.BodyKey, media.JSONKey, schema.Key, "name"),
+			expected: "ValidationFixtureEndpointPostRequestBodyModelName",
+		},
+	}
```

Comment:
Pure bliss. well done

Reply:
These have literally saved me hours in debugging time